vulnerabilities/models.py

```diff
@@ -2744,6 +2748,73 @@ def url(self):
             return f"https://github.com/nodejs/security-wg/blob/main/vuln/npm/{id}.json"
 
 
+class Patch(models.Model):
+    """
+    A generic patch linked to an advisory.
+    Used for raw text patches, fallback scenarios, or unsupported VCS types.
+    """
+
+    patch_url = models.URLField(
+        null=True,
+        blank=True,
+        help_text="URL to the patch file or diff.",
+    )
+
+    patch_text = models.TextField(
+        null=True,
+        blank=True,
+        help_text="The actual text content of the code patch/diff.",
+    )
+
+    patch_checksum = models.CharField(
+        max_length=128,
+        blank=True,
+        null=True,
+        help_text="SHA512 checksum of the patch content.",
+    )
+
+    def clean(self):
+        if not self.patch_url and not self.patch_text:
+            raise ValidationError("Either patch_url or patch_text must be provided.")
+
+    def save(self, *args, **kwargs):
+        # https://docs.djangoproject.com/en/4.2/ref/models/instances/#django.db.models.Model.clean
+        self.full_clean()
```

Comment:
Also this is not needed for database level constraint

Reply:
you mean check constraint like this https://docs.djangoproject.com/en/5.2/ref/models/constraints/#checkconstraint

    class Meta:
        constraints = [
            models.CheckConstraint(check=(Q(patch_url__isnull=False) | Q(patch_text__isnull=False)),
                name="patch_url_or_patch_text",
            )
        ]

Reply:
yes